lib/puppet/parser/functions/cidr_to_range.rb

```diff
@@ -0,0 +1,30 @@
+module Puppet::Parser::Functions
+
+  newfunction(:cidr_to_range, :type => :rvalue, :doc => <<-'ENDHEREDOC'
+    Converts an CIDR address of the form 192.168.0.1/24 into a range of IP address excluding the network and broadcast
+    addresses.
+
+    ENDHEREDOC
+    ) do |args|
+
+    unless args.length == 1
+      raise Puppet::ParseError, ("cidr_to_range(): wrong number of arguments (#{args.length}; must be 1)")
+    end
+
+    arg = args[0]
+    unless arg.respond_to?('to_s')
```

Comment:
Thanks, yeh that was a bit useless. I've replaced it with a stricter `String` check as any other data type coming into this function would be invalid.